### PR TITLE
 feat(app, protocol-designer, step-generation): schema v8 migration

### PR DESCRIPTION
**Pull request by jerader**
closes RAUT-828

# Overview

This PR migrated the types in PD, step-generation, and the app to schema v8. 

# Test Plan

Create a protocol in PD without any deck configuration (no waste chute, staging area slots) and download. Upload it to the app. The app should function as expected and the protocol should say creation method PD 8.0.

Screen Shot 2023-10-31 at 2 03 53 PM

# Changelog

- export v8 types in the index file in shared-data and update all instances of v7 to v8 except for when it isn't needed.
- deprecate usage of `ProtocolAnalysisFile`
- update `fileCreator` in PD to match the new schema shape
- update 7_1_0 PD migration file to say 8_0_0 instead for a v8 migration.

Note: PD migration cypress test will be updated/thoroughly tested in a follow up PR!

# Review requests

see test plan

# Risk assessment

low

### CODE REVIEW

File: shared-data/js/helpers/parseProtocolData.ts [hunk 1]
```diff
@@ -125,12 +120,6 @@ export function validateJsonProtocolFileContents(
   }
 }
 
-export function protocolHasLiquids(
-  protocol: ProtocolAnalysisFile<{}> | ProtocolAnalysisOutput
-): boolean {
-  return 'liquids' in protocol && size(protocol.liquids) > 0
-}
-
```
Comment (jerader) [inline shared-data/js/helpers/parseProtocolData.ts:133]: this util isn't being used anymore

**Review by sfoster1 — APPROVED** [review]
Looks good to me and fixes an issue where I couldn't add a new command to command schema 8 without things breaking!

**Comment by sfoster1** [comment]
Removed deck schema specification on request of rss here

**Review by smb2268 — APPROVED** [review]
No red flags for me! I tested this with a simple protocol and upload to the app worked as expected on a dev app build on this branch.

**Pull request merged**